lib/spack/spack/build_environment.py

```diff
@@ -248,7 +248,7 @@ def set_build_environment_variables(pkg, env, dirty=False):
         dirty (bool): Skip unsetting the user's environment settings
     """
     # Gather information about various types of dependencies
-    build_deps      = pkg.spec.traverse(root=False, deptype=('build'))
+    build_deps      = pkg.spec.dependencies(deptype='build')
```

Comment:
Shouldn't run-time dependencies of direct build-time dependencies also be added?